### DISCUSSION TITLE
Cmd popups when I use exec, -ldflags -H=windowsgui not effective

### DISCUSSION THREAD

**Discussion by sithembiso** [wailsapp/wails#1734] [Q&A] [answered]
I've been fiddling with this for quite some time. I'm not sure if it's a Wails issue or not, hence raising the question here. I have an app that relies on a background process that I trigger using Go's `exec.Command("cmd", "/c", "start", "app.exe")`. Every time I run the app though, I get these terminal popups. So, I have modified my build command to:

```
wails build -ldflags -H=windowsgui
```
The output looks like this:
```
Wails CLI v2.0.0-beta.43

App Type:               desktop
Platforms:              windows/amd64
Compiler:               C:\Program Files\Go\bin\go.exe
Build Mode:             Production
Skip Frontend:          false
Compress:               false
Package:                true
Clean Build Dir:        false
LDFlags:                "-H windowsgui"
Tags:                   []
Race Detector:          false

Building target: windows/amd64
------------------------------
  - Installing frontend dependencies: Done.
  - Compiling frontend: Done.
  - Generating bundle assets: Done.
  - Compiling application: Done.
```
I've looked at other issues, mostly from Wails v1.0, but nothing has been helpful yet. 
Any input will be highly appreciated.

**Comment by leaanthony** [comment]
If you spawn CMD then you'll get a window regardless. Is there a reason you aren't using the `exec` package? 🙏

**Reply by sithembiso** [reply]
I started using CMD because I was getting the same issue with running the command directly. Before I was doing `exec.Command("app.exe")`. Same issue.

**Comment by sithembiso** [comment, marked as answer]
These options on Windows saved the day:
```go
cmd := exec.Command("app.exe")
cmd.SysProcAttr = &syscall.SysProcAttr{
	HideWindow:    true,
	CreationFlags: 0x08000000,
}
cmd.Start()
```

Source: [https://stackoverflow.com/a/48365926/789655](https://stackoverflow.com/a/48365926/789655)
